lib/p5.speech.js

```diff
@@ -166,38 +136,66 @@
   // (using voices found in the voices[] array), or by index.
   p5.Speech.prototype.setVoice = function(_v) {
     // type check so you can set by label or by index:
-    if(typeof(_v)=='string') this.utterance.voice = this.voices.filter(function(v) { return v.name == _v; })[0];
-    else if(typeof(_v)=='number') this.utterance.voice = this.voices[Math.min(Math.max(_v,0),this.voices.length-1)];
+    if(typeof(_v)=='string') this.voice = this.voices.filter(function(v) { return v.name == _v; })[0];
+    else if(typeof(_v)=='number') this.voice = this.voices[Math.min(Math.max(_v,0),this.voices.length-1)];
   };
 
   // volume of voice. API range 0.0-1.0.
   p5.Speech.prototype.setVolume = function(_v) {
-    this.utterance.volume = Math.min(Math.max(_v, 0.0), 1.0);
+    this.volume = Math.min(Math.max(_v, 0.0), 1.0);
   };
 
   // rate of voice.  not all voices support this feature.
   // API range 0.1-2.0.  voice will crash out of bounds.
   p5.Speech.prototype.setRate = function(_v) {
-    this.utterance.rate = Math.min(Math.max(_v, 0.1), 2.0);
+    this.rate = Math.min(Math.max(_v, 0.1), 2.0);
   };
 
   // pitch of voice.  not all voices support this feature.
   // API range >0.0-2.0.  voice will crash out of bounds.
   p5.Speech.prototype.setPitch = function(_v) {
-    this.utterance.pitch = Math.min(Math.max(_v, 0.01), 2.0);
+    this.pitch = Math.min(Math.max(_v, 0.01), 2.0);
   };
 
   // sets the language of the voice.
   p5.Speech.prototype.setLang = function(_lang) {
-    this.utterance.lang = _lang;
+    this.lang = _lang;
 }
 
   // speak a phrase through the current synthesizer:
   p5.Speech.prototype.speak = function(_phrase) {
-    if(this.interrupt) this.synth.cancel();
-    this.utterance.text = _phrase;
+    if(this.interrupt) this.cancel();
+    var utterance = this.newUtterance()
+    utterance.text = _phrase;
+
+    this.synth.speak(utterance);
+  };
+
+  var speechToUtteranceProperties = {
+    interrupt: "interrupt",
+    onLoad: "onload",
+    onStart: "onstart",
+    onPause: "onpause",
+    onResume: "onresume",
+    onEnd: "onend",
```

Comment:
I don't think these events need to be passed down - they bubble up to the SpeechSynthesiser instance automatically?